x-pack/platform/plugins/shared/streams_app/public/components/group_stream_modification_flyout/dashboards_tab.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiButtonIcon, EuiFieldText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+import type { TabProps } from './types';
+import { useKibana } from '../../hooks/use_kibana';
+
+export function DashboardsTab({ formData, setFormData }: TabProps) {
+  const {
+    dependencies: {
+      start: { dashboard: dashboardStart },
+    },
+  } = useKibana();
+
+  useEffect(() => {
+    setSelectedItems(formData.dashboards);
+  }, [formData.dashboards]);
+  const [selectedItems, setSelectedItems] = useState<{ id: string; title: string }[]>(
+    formData.dashboards
+  );
```

Comment:
Since this state changes only when `formData.dashboard` property changes, there is no use for this local state and side effect, you can directly use the property value:
```ts
const selectedItems = formData.dashboards
```